mission/functions/systems/actions/fn_action_gather_intel.sqf

```diff
@@ -1,19 +1,66 @@
+private _object = player;
+private _title = "Gather Enemy Intel";
+private _iconIdle = "custom\holdactions\holdAction_documents_ca.paa";
+private _iconProgress = "custom\holdactions\holdAction_documents_ca.paa";
+
+
+// this is a hacky workaround following similar logic to this github gist
+// https://gist.github.com/CalebSerafin/f4a16f2f564a438882ca6f32107bec24
+// which tries to solve this issue
+// https://feedback.bistudio.com/T155638
+//
+// however, instead of relying on undocumented functionality
+// (which could disappear in the future!) i've changed it so that the
+// holdAction UI is always redrawn every 5-ish seconds.
+
+private _conditionToShow = str {
+	call {
+		private _needToReDrawHoldAction = (serverTime - 5) > (player getVariable ["vn_mf_bn_actions_intel_client_lastcheck", 0]);
+		private _actualCondition = (
+			player distance cursorObject < 5
+			&& {typeOf cursorObject in ['Land_Map_unfolded_F','Land_Map_unfolded_Malden_F', 'vn_b_prop_cabinet_02']}
+		);
+		private _res = false;
+		// don't need to redraw yet and we're looking at an intel thing -- show the UI.
+		if (!_needToReDrawHoldAction && _actualCondition) then {_res = true};
```

Comment:
if 5 seconds not passed and is valid object within correct distance, show the UI